src/core/gui/PageView.h

```diff
@@ -188,6 +190,26 @@ class XojPageView: public Redrawable, public PageListener {
 
     void drawLoadingPage(cairo_t* cr);
 
+    /**
+     * @brief Make and display a popover dialog near the given location.
+     *
+     * @param rect specifies the location of the dialog.
+     * @param child is added to the dialog before displaying.
+     * @returns a pointer to the popover dialog.
+     */
+    GtkWidget* makePopover(const XojPdfRectangle& rect, GtkWidget* child);
+
+    /**
+     * @brief Display a popover with link-related actions, if one
+     *  is at a given location on the page.
+     *
+     * @param page is the current page
+     * @param targetX
+     * @param targetY the link must contain (targetX, targetY)
+     * @returns true iff a URI link exists near/at (targetX, targetY) => a popover was shown
+     */
+    bool displayLinkPopover(std::shared_ptr<XojPdfPage> page, double targetX, double targetY);
```

Comment:
I think this function could be made static (with a few modifications in the implementation).